metricbeat/beater/event.go

```diff
@@ -0,0 +1,147 @@
+package beater
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/filter"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+const (
+	defaultType = "metricsets"
+)
+
+// eventBuilder is used for building MetricSet events. MetricSets generate a
+// data in the form of a common.MapStr. This builder transforms that data into
+// a complete event and applies any Module-level filtering.
+type eventBuilder struct {
+	moduleName    string
+	metricSetName string
+	host          string
+	startTime     time.Time
+	fetchDuration time.Duration
+	event         common.MapStr
+	fetchErr      error
+	filters       *filter.FilterList
+	metadata      common.EventMetadata
+}
+
+// build builds an event from MetricSet data and applies the Module-level
+// filters.
+func (b eventBuilder) build() (common.MapStr, error) {
+	// event may be nil when there was an error fetching.
+	event := b.event
+	if event == nil {
+		event = common.MapStr{} // TODO (akroh): do we want to send an empty event field?
```

Comment:
Could this affect some visualisations in Kibana or some filters? Not sure what the pro / con are of "not having a field" vs "having an empty field". @simianhacker Any opinions on that?